DevGuideExamples/DCPS/Messenger/CMakeLists.txt

```diff
@@ -3,8 +3,6 @@ cmake_minimum_required(VERSION 3.8.2)
 
 find_package(OpenDDS REQUIRED)
 
-set(CMAKE_CXX_COMPILER ${OPENDDS_COMPILER})
```

Comment:
To make this work for systems that don't use the default compiler, the script that invokes `cmake` needs to pass `-DCMAKE_CXX_COMPILER=<something>`.